src/header/common/strict_transport_security.rs

```diff
@@ -85,35 +151,56 @@ impl FromStr for StrictTransportSecurity {
-                    (Some(left), Some(right))
-                    if unicase::eq_ascii(left.trim(), "max-age") => {
-                        right
-                            .trim()
-                            .trim_matches('"')
-                            .parse()
-                            .map(Directive::MaxAge)
-                    },
-                    _ => Ok(Directive::Unknown)
+            .map(|sub| match sub {
+                sub if unicase::eq_ascii(sub, "includeSubDomains") => 
+                    Ok(Directive::IncludeSubdomains),
+                sub if unicase::eq_ascii(sub, "preload") => 
+                    Ok(Directive::Preload),
+                sub => {
+                    let mut sub = sub.splitn(2, '=');
+                    match (sub.next(), sub.next()) {
+                        (Some(left), Some(right))
+                        if unicase::eq_ascii(left.trim(), "max-age") => {
+                            right
+                                .trim()
+                                .trim_matches('"')
+                                .parse()
+                                .map(Directive::MaxAge)
+                        },
+                        _ => Ok(Directive::Unknown)
+                    }
                 }
             })
-            .fold(Ok((None, None)), |res, dir| match (res, dir) {
-                (Ok((None, sub)), Ok(Directive::MaxAge(age))) => Ok((Some(age), sub)),
-                (Ok((age, None)), Ok(Directive::IncludeSubdomains)) => Ok((age, Some(()))),
-                (Ok((Some(_), _)), Ok(Directive::MaxAge(_))) |
-                (Ok((_, Some(_))), Ok(Directive::IncludeSubdomains)) |
-                (_, Err(_)) => Err(::Error::Header),
-                (res, _) => res
+            .fold(Ok((None, None, None)), |res, dir| match (res, dir) {
+                ( Ok((None, subd, prel)), Ok(Directive::MaxAge(age))) => Ok((Some(age),subd,prel)),
+                ( Ok((mage, None, prel)), Ok(Directive::IncludeSubdomains)) => Ok((mage,Some(true),prel)),
+                ( Ok((mage, subd, None)), Ok(Directive::Preload)) => Ok((mage,subd,Some(true))),
+                ( Ok((Some(_), _, _)), Ok(Directive::MaxAge(_))) |
+                ( Ok((_, Some(_), _)), Ok(Directive::IncludeSubdomains)) |
+                ( Ok((_, _, Some(_))), Ok(Directive::Preload)) |
+                ( _, Err(_) ) => Err(::Error::Header),
+                ( res, _ ) => res
             })
             .and_then(|res| match res {
-                (Some(age), sub) => Ok(StrictTransportSecurity {
+                (Some(age), sub, pre) if pre.is_none() => Ok(StrictTransportSecurity{
                     max_age: age,
-                    include_subdomains: sub.is_some()
+                    include_subdomains: sub.is_some(),
+                    preload: false
                 }),
+                (Some(age), sub, pre) if pre.is_some() => {
+                    if sub.is_none() || age < PRELOAD_MIN_AGE {
```

Comment:
I wouldn't enforce the limit when parsing the header either. It's a perfectly valid header, it just doesn't meet the requirements for Google's list at the time of writing (but could in the future).